### PR TITLE
Add support for pseudo elements

### LINKED ISSUE #99: Add support for pseudo-elements
In Clay: https://github.com/sebastiaanvisser/clay/blob/54dc9eaf0abd180ef9e35d97313062d99a02ee75/src/Clay/Selector.hs#L151

### PR DESCRIPTION
**Pull request by vyorkin**
Resolves #99 

If I get it right:
`::before` and `::after` is the same as `:before` and `:after` (aliases)

image

### CODE REVIEW

**Comment by vyorkin** [comment]
Rebased, ready for review

**Review by puffnfresh — APPROVED** [review]

**Comment by garyb** [comment]
This maybe seems like one of those cases where we can have an exhaustive enumeration of `Pseudo` values?

**Comment by vyorkin** [comment]
Good point! I'll update the PR

File: src/CSS/Selector.purs [hunk 1]
```diff
@@ -1,42 +1,9 @@
 module CSS.Selector where
 
 import Prelude
-
-import Data.String (take, drop)
-
+import CSS.Refinement (Refinement(..), byClass, byId)
 import CSS.String (class IsString)
-
-data Predicate
```
Comment (vyorkin) [inline src/CSS/Selector.purs]: extracted to separate module: `CSS.Predicate`

**Comment by vyorkin** [comment]
I don't know if we really need that `IsString Refinement` instance, so I'll make another commit that removes it and also removes these `Raw String` data constructors.

**Comment by nsaunders** [comment]
Hi @vyorkin, apologies that this was never merged in. As you might know, I did inherit maintainership of this library about a year ago; but, to be frank, I found that it needs so much work (so many incompatible design changes) to be practically useful that a better use of my time and effort was just to start over from scratch: See [Tecton](https://github.com/nsaunders/tecton). Assuming you're still interested in statically-typed CSS, I think you'll find that library to be much more feature-complete than this one (including pseudo-elements); but contributions are also welcome in case you find anything you need to be missing.

**Pull request closed without merging**